frontend/src/metabase/search/containers/SearchApp.jsx

```diff
@@ -25,172 +30,90 @@ import {
-    setFilter(filterItem && filterItem.filter);
-    setPage(0);
-  };
+  const searchText = useMemo(
+    () => getSearchTextFromLocation(location),
+    [location],
+  );
 
-  const query = {
-    q: location.query.q,
-    limit: PAGE_SIZE,
-    offset: PAGE_SIZE * page,
-  };
+  const searchFilters = useMemo(() => {
+    return getFiltersFromLocation(location);
+  }, [location]);
+
+  const [selectedSidebarType, setSelectedSidebarType] = useState(null);
+
+  useEffect(() => {
+    setSelectedSidebarType(null);
+  }, [searchText, searchFilters]);
+
+  const query = useMemo(
+    () => ({
+      q: searchText,
+      ..._.omit(searchFilters, SearchFilterKeys.Type),
+      models: selectedSidebarType ?? searchFilters[SearchFilterKeys.Type],
+      limit: PAGE_SIZE,
+      offset: PAGE_SIZE * page,
+    }),
+    [searchText, searchFilters, selectedSidebarType, page],
+  );
 
-  if (filter) {
-    query.models = filter;
-  }
+  const onChangeSelectedType = filter => {
+    setSelectedSidebarType(filter);
+    setPage(0);
+  };
 
   return (
-    <SearchRoot>
-      {location.query.q && (
+    <SearchRoot data-testid="search-app">
+      {searchText && (
         <SearchHeader>
-          <Subhead>{jt`Results for "${location.query.q}"`}</Subhead>
+          <Subhead>{jt`Results for "${searchText}"`}</Subhead>
         </SearchHeader>
       )}
-      <div>
-        <Search.ListLoader query={query} wrapped>
-          {({ list, metadata }) => {
-            if (list.length === 0) {
-              return (
-                <SearchEmptyState>
-                  <Card>
-                    <EmptyState
-                      title={t`Didn't find anything`}
-                      message={t`There weren't any results for your search.`}
-                      illustrationElement={<img src={NoResults} />}
-                    />
-                  </Card>
-                </SearchEmptyState>
-              );
-            }
-
-            const availableModels = metadata.available_models || [];
-
-            const filters = SEARCH_FILTERS.filter(f =>
-              availableModels.includes(f.filter),
-            );
-
-            return (
-              <SearchBody>
-                <SearchMain>
-                  <Fragment>
-                    <SearchResultSection items={list} />
-                    <div className="flex justify-end my2">
-                      <PaginationControls
-                        showTotal
-                        pageSize={PAGE_SIZE}
-                        page={page}
-                        itemsLength={list.length}
-                        total={metadata.total}
-                        onNextPage={handleNextPage}
-                        onPreviousPage={handlePreviousPage}
-                      />
-                    </div>
-                  </Fragment>
-                </SearchMain>
+      <Search.ListLoader query={query} wrapped>
+        {({ list, metadata }) =>
+          list && list.length > 0 ? (
```

Comment:
```suggestion
          list?.length > 0 ? (
```
should be more concise